newton/_src/sim/model.py

```
@@ -813,68 +815,62 @@ def set_gravity(
-            self._collision_pipeline = CollisionPipeline.from_model(
-                model=self,
-                rigid_contact_max_per_pair=rigid_contact_max_per_pair,
-                soft_contact_max=soft_contact_max,
-                soft_contact_margin=soft_contact_margin,
-                edge_sdf_iter=edge_sdf_iter,
-                requires_grad=requires_grad,
+        if self._collision_pipeline is None:
+            raise ValueError(
+                "Model does not have a collision pipeline. Call model.contacts() "
+                "or use a your collision pipeline directly: CollisionPipeline.collide(contacts)."
             )
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)